app/assets/javascripts/components/story/StoryAttachment.js

```diff
@@ -58,7 +62,43 @@ class StoryAttachment extends React.Component {
     };
   }
 
+  componentWillMount() {
+    this.setAttachinaryOptions();
+  }
+
+  setAttachinaryOptions() {
+    this.getUpdatedSignature()
+      .then((response) => {
+        this.setState({
+          attachinaryOptions: response,
+        })
+      })
+  }
+
+  updateSignature() {
+    latestTimestamp = this.state.attachinaryOptions.html.data.form_data.timestamp * 1000;
+    currentTimestamp = new Date().getTime();
+    elapsedTime = currentTimestamp - latestTimestamp;
+    timeToInvalidate = 3600000;
+    if(elapsedTime >= timeToInvalidate){
+      this.setAttachinaryOptions();
+    }
+  }
+
+  getUpdatedSignature() {
+    const options = {
```

Comment:
Maybe it would be a good idea to extract it to a separate class/object to make it reusable, something like a model called `AttachmentSignature`, what do you think? It doesn't need to be an actual Backbone model but it can have the same interface as one, like a static method `AttachmentSignature.fetch()`.

Reply:
I am on it now